components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -238,6 +247,15 @@ public String buildIDToken(OAuthAuthzReqMessageContext authzReqMessageContext,
         AuthenticatedUser authorizedUser = authzReqMessageContext.getAuthorizationReqDTO().getUser();
         String subject =
                 getSubjectClaim(authzReqMessageContext, tokenRespDTO, clientId, spTenantDomain, authorizedUser);
+        String callbackUri;
+        try {
+            callbackUri = OIDCClaimUtil.getCallbackUrl(clientId);
+        } catch (InvalidOAuthClientException e) {
+            throw new IdentityOAuth2Exception("Error while getting the registered callback URI for client_id: " +
```

Comment:
Since the caught exception is a client exception, lets throw an IdentityOAuth2ClientException which seems to be the client type exception for IdentityOAuth2Exception